src/Symfony/Component/DependencyInjection/Argument/AbstractArgument.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\DependencyInjection\Argument;
+
+/**
+ * Represents an abstract parameter, which have to be replaced by one of CompilerPasses.
+ */
+final class AbstractArgument
+{
+    /** @var string */
+    private $serviceId;
+
+    /** @var string */
+    private $argKey;
+
+    /** @var string */
+    private $text;
+
+    public function __construct(string $serviceId, string $argKey, string $text)
+    {
+        $this->serviceId = $serviceId;
+        $this->argKey = $argKey;
+        $this->text = $text;
+    }
+
+    /**
+     * @return string Error text
+     */
+    public function __toString(): string
+    {
+        return sprintf(
+            'Argument "%s" is not replaced for service "%s" yet: %s',
```

Comment:
`Argument "%s" must be set for service "%s": %s.`
but when $text is empty, we could remove the colon

Reply:
fixed